wpgraphql-ide.php

```diff
@@ -50,6 +50,29 @@ function user_lacks_capability(): bool {
     return ! current_user_can( $capability_required );
 }
 
+/**
+ * Checks if the current page is intended for the dedicated WPGraphQL IDE (non-drawer).
+ *
+ * @return bool True if the current page is a dedicated WPGraphQL IDE page, false otherwise.
+ */
+function is_dedicated_ide_page(): bool {
+    if ( ! function_exists( 'get_current_screen' ) ) {
+        return false;
+    }
```

Comment:
Should we provide a warning for developers here?

```
_doing_it_wrong( __FUNCTION__, 'Function should only be called within admin pages context.', '1.0.1' );
```

Reply:
I don't think so. The `is_dedicated_ide_page()` function is being used to pass initial context via localized json, and that will execute on both the front-end and admin, so a `doing_it_wrong()` response would mean anytime a front-end page is loaded we'd see the doing_it_wrong message, when there's nothing wrong with asking if the current page is the dedicated ide page or not.